indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1882,33 +1882,35 @@ TransactionalSegmentPublisher createPublisher(TaskToolbox toolbox, boolean useTr
   }
 
   private boolean verifyInitialRecordAndSkipExclusivePartition(
-      final OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType> record,
-      final Map<PartitionIdType, SequenceOffsetType> intialSequenceSnapshot
+      final OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType> record
   )
   {
-    if (intialSequenceSnapshot.containsKey(record.getPartitionId())) {
-      if (record.getSequenceNumber().compareTo(intialSequenceSnapshot.get(record.getPartitionId())) < 0) {
-        throw new ISE(
-            "Starting sequenceNumber [%s] does not match expected [%s] for partition [%s]",
-            record.getSequenceNumber(),
-            intialSequenceSnapshot.get(record.getPartitionId()),
-            record.getPartitionId()
+    // Check only for the first record among the record batch.
+    if (initialOffsetsSnapshot.contains(record.getPartitionId())) {
+      final SequenceOffsetType currOffset = currOffsets.get(record.getPartitionId());
+      if (currOffset != null) {
```

Comment:
When is `currOffset` null? It seems to defeat the purpose of this check if we can get a `record` to check, and then don't check it because we don't know what the current offset is supposed to be.

Reply:
Hmm, maybe it's better to throw an error if it's null. Will raise a PR.

Reply:
Thanks